crates/types/src/blockchain/header/v1.rs

```diff
@@ -21,22 +24,22 @@ use crate::{
 #[cfg_attr(any(test, feature = "test-helpers"), derive(Default))]
 pub struct BlockHeaderV1 {
     /// The application header.
-    pub application: ApplicationHeader<GeneratedApplicationFields>,
+    pub(crate) application: ApplicationHeader<GeneratedApplicationFieldsV1>,
     /// The consensus header.
-    pub consensus: ConsensusHeader<GeneratedConsensusFields>,
+    pub(crate) consensus: ConsensusHeader<GeneratedConsensusFields>,
```

Comment:
Why do we change the visibility for the application and consensus headers?
Also, modifying these fields makes this change breaking.

Reply:
they should have never been public in the first place :)
it removes the point of having an enum for versioned headers

this is a breaking change, it is intended